base_layer/core/src/mining/blake_miner.rs

```diff
@@ -74,6 +73,8 @@ impl CpuBlakePow {
                 info!(target: LOG_TARGET, "Mining hash rate per thread: {:.6} MH/s", hash_rate);
                 last_measured_nonce = nonce;
                 start = Instant::now();
+
+                header.timestamp = EpochTime::now();
```

Comment:
I still feel this might break mining as there is now a change that you never will guess the nonce. 
Either remove this or the nonce increase. With each iteration you change two variables. I agree you dont "search" a space. But by changing both each iteration you need to through 2 lucky coins and not one.

Reply:
https://en.wikipedia.org/wiki/Gambler's_fallacy

Reply:
https://en.bitcoin.it/wiki/Block Read the "What if I'm 1% towards calculating a block and...?"